UI Improvements

botorabi opened the pull request:
Several improvements for front-end and back-end, preparing version 0.8.0.

botorabi committed:
Added more control to team lead role; improved UI

- Team lead role can now create and edit tasks and report mail configuration
- Added more back-end access control to team and task related operations
- Extended and improved front-end

botorabi committed:
Added unit-test for some report generation classes

botorabi committed:
Corrected the "current week" number

- Now a calendar week is considered to begin on Monday

botorabi committed:
Added more unit-tests; improved UI

botorabi committed:
Began the work on translation support for the front-end

botorabi committed:
Added more translations, improved the Translator code

botorabi committed:
Added paged fetching of progress entries

- Make use of back-end's paged fetch of progress entries

botorabi committed:
Fixed a problem with sending report if more that one recipient was defined

botorabi committed:
Added session timeout handler, fixed an issue in role widget, improved Config

botorabi committed:
Added a footer to all pages, additional German translations


- Footer contains app name and version, login name and timeout for automatic logout
- Localization: Moved 'Version' form NavDrawer context to Common context
- Set the automatic logout timeout to 30 minutes

botorabi committed:
Improved add/remove buttons; fixed an issue with UTF8 name display

- Fixed a display problem of UTF8 strings in report team leader names

botorabi committed:
Improved UI

- Replaced expansion widgets by tabs in admin and team lead page

botorabi committed:
Added an About page and updated the app icons

eparo commented:
1. "About..." ist in der NavBar doppelt enthalten
2. Angelegte Aufgaben werden nicht angezeigt

eparo commented:
3. "Edit Users" zeigt keine Häkchen bei den Rollen an

botorabi committed:
Improved UI, fixed a problem with retrieving tasks as admin user

- Admin user has access to all tasks now
- Added all related team names to task entry
- Improved the column spacing in tables
- Fixed a problem with displaying user roles
- Fixed a problem with displaying the About entry twice in navigation drawer

eparo committed:
Fix quotes in translation strings

eparo committed:
Some translation corrections

eparo reviewed (approved):
Sieht jetzt OK aus.
